### PR TITLE
workflows.sh script to perform various workflow operations

### PR DESCRIPTION
**Pull request by NateThornton**
```
Run various commands against DWS workflows

Usage: ./workflows.sh COMMAND [ARGS...]

Commands:
    advance WORKFLOW [--no-wait]        Advance the desired state of a workflow to the next state. If
                                        the workflow is not currently in a ready state, wait up to 60s
                                        for the workflow to become ready prior to advancing state. Use
                                        the '--no-wait' option to bypass the wait check and fail if the
                                        workflow is not currently ready.

    teardown [WORKFLOW]... [--all]      Set the desired state on a series of workflows to Teardown.
                                        This does not guarantee the Teardown state is achieved.
                                        Use the '--all' option to teardown all workflows in a system.
```

### CODE REVIEW

**Comment by roehrich-hpe** [comment]
You're re-imagining dwsutil?

**Comment by NateThornton** [comment]
> You're re-imagining dwsutil?

What never should have existed is no more - I archived dwsutil last month

File: workflows.sh [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        WORKFLOW=${2}
+        if [ ${!#} = "--no-wait" ]; then
+            if [ "$(kubectl get workflow.dws/"$WORKFLOW" --output=custom-columns=:.status.ready --no-headers)" = "false" ]; then
+                echo "ERROR: Workflow $WORKFLOW not ready. Aborting." >&2
+                exit 1
+            fi
+        fi
+
+        TIMEOUT=60s
+        if ( timeout $TIMEOUT kubectl get workflow.dws/"$WORKFLOW" --output=custom-columns=:.status.ready --no-headers --watch & ) | grep -q "true"; then
```
Comment (matthew-richerson) [inline workflows.sh:57]: This should check whether spec.desiredState == status.State before looking at ready. Probably doesn't matter for this script if it's only going to be used internally.

**Review by matthew-richerson — APPROVED** [review]

File: workflows.sh [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+                                        Use the '--all' option to teardown all workflows in a system.
+EOF
+}
+
+if [ $# -lt 1 ]; then
+    usage
+    exit 1
+fi
+
+declare -A NEXTSTATEARRAY=(["Proposal"]="Setup" ["Setup"]="DataIn" ["DataIn"]="PreRun" ["PreRun"]="PostRun" ["PostRun"]="DataOut" ["DataOut"]="Teardown" ["Teardown"]="Teardown")
```
Comment (bdevcich-hpe) [inline workflows.sh:44]: This doesn't seem to work in osx:
```
./workflows.sh advance nnf08-global-lustre
./workflows.sh: line 44: declare: -A: invalid option
declare: usage: declare [-afFirtx] [-p] [name[=value] ...]
./workflows.sh: line 57: timeout: command not found
ERROR: Workflow nnf08-global-lustre not ready after 60s. Aborting.
```

File: workflows.sh (hunk 1 above)
Comment (NateThornton) [reply workflows.sh:57]: That would complicate the timeout logic. Any idea how to implement that?

File: workflows.sh (hunk 2 above)
Comment (NateThornton) [reply workflows.sh:44]: what terminal are you using?

**Comment by ajfloeder** [comment]
This looks to be the CLI UI to smooth out kubectl issues. A `list` command would be nice just to see the workflows in your current context.

**Review by ajfloeder — APPROVED** [review]

**Pull request closed without merging**